Extend support for display names

Corniel opened the pull request:
Qowaiv supports email addresses with a display name: 
* `Joe Smith <email@qowaiv.org>`
* `(Joe Smith) email@qowaiv.org` (in fact as comment)

The change includes:
* `"Joe Smith" email@qowaiv.org`

Questions to answer: 
* Should also `'Joe Smith' email@qowaiv.org` be supported?
* Which escape character(s) for the quote should be supported, if any?
* Should a space character after the end quote be mandatory?

See: #68

Corniel commented:
@salmarko Thanks for the approval, but those 3 questions still have to be answered.

salmarko commented:
_Should also 'Joe Smith' email@qowaiv.org be supported?_ (No, as far as I am concerned)
_Which escape character(s) for the quote should be supported, if any?_ (Only \ )
_Should a space character after the end quote be mandatory?_ (Yes)

Corniel commented:
@salmarko :
* Added support for '\' escape character.
* Made whitespace character afterward mandatory
* Added `WithDisplayName(string)` method to convert back to something with a display name. (always Display name <emailaddress>)
